# Sensitivity Analysis: Sex Stratified

The following analysis include running PACC \~ GeneExp\*Ab in only males and only females in the genes that survived FDR correction in the PACC \~ GeneExp\*Sex\*Ab analyses.

```{r setup, include = FALSE, warning = FALSE, message = FALSE}
library(plyr)
library(nlme)
library(ggplot2)
library(sjPlot)
library(dplyr)
library(gtsummary)
library(flextable)
library(ggrepel)
library(readr)
library(DT)
library(tidyr)

knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
theme_set(theme_bw())
```

```{r}
load(file = "~/Dropbox (Partners HealthCare)/Mabel/Longitudinal A4 & Gene Expression/A4_GeneExp_Project/Data/RNAseqData_Autosomal_XLinked_wDose_121124.RData")

gene_names <- readRDS("~/Dropbox (Partners HealthCare)/Mabel/Longitudinal A4 & Gene Expression/A4_GeneExp_Project/Data/A4_RNAseq_GeneNames_081424.rds")

## PACC Models
load("~/Dropbox (Partners HealthCare)/Mabel/Longitudinal A4 & Gene Expression/A4_GeneExp_Project/Data/PACC Models/LME_PACC_GeneSexAb_Models_2024-12-06.RData")
```

```{r}
gene_names <- select(gene_names, gene = ensembl_gene_id, chromosome_name, external_gene_name)

bl_amyloid <- rna %>%
  filter(!is.na(suvr_cer)) %>%
  group_by(BID) %>%
  arrange(BID, pacc_time) %>%
  slice(1) %>%
  select(BID, bl_Amyloid = suvr_cer, bl_AmyloidStatus = overall_score)

rna <- merge(rna, bl_amyloid, by = "BID", all.x = TRUE)

```

```{r}
model_loops <- function(data, genes, outcome, predictor, covariates, main, tab_name, folder, mod_type){
  ModelTable <- data.frame(gene = character(length(genes)), beta = numeric(length(genes)), se = numeric(length(genes)), 
                           df = numeric(length(genes)), t = numeric(length(genes)), p = numeric(length(genes)))
  
  for (i in 1:length(genes)){
    modelSum <- summary(lme(reformulate(c(paste0(genes[i], predictor), covariates), response = outcome), 
                            data = data, random =~ pacc_time | BID, na.action = "na.omit",
                            control = list(opt = 'optim', msMaxIter = 1000, msMaxEval = 1000)
    ))
    ModelTable[i,2:6] <- modelSum$tTable[which(rownames(modelSum$tTable) == paste0(genes[i], main)),]
    ModelTable[i,1] <- genes[i]
  }
  ModelTable$fdr_p <- NA
  ModelTable$outcome <- outcome
  assign(tab_name, ModelTable, .GlobalEnv)
  #save(list = tab_name, file = paste0("~/Dropbox (Partners HealthCare)/Mabel/Longitudinal A4 & Gene Expression/", folder, "/LME_", mod_type, "_Models_", Sys.Date(), ".RData"))
}
```

```{r}
model_loops(data = subset(rna, SEX == "Female"), 
            genes = subset(PACCTable_GeneSexAb_wDose, fdr_p < 0.05)$gene,
            outcome = "PACC",
            predictor = "*bl_Amyloid*pacc_time",
            covariates = c("bl_pacc_age*pacc_time", "EDCCNTU*pacc_time",
                           "Cohort*pacc_time", "QSVERSION", "Cumulative_Dose_Scaled"),
            main = ":bl_Amyloid:pacc_time",
            tab_name = "tmp_fem",
            folder = "PACC Models",
            mod_type = "PACC_GeneAb")

tmp_fem$Subset <- "Females"

model_loops(data = subset(rna, SEX == "Male"), 
            genes = subset(PACCTable_GeneSexAb_wDose, fdr_p < 0.05)$gene,
            outcome = "PACC",
            predictor = "*bl_Amyloid*pacc_time",
            covariates = c("bl_pacc_age*pacc_time", "EDCCNTU*pacc_time",
                           "Cohort*pacc_time", "QSVERSION", "Cumulative_Dose_Scaled"),
            main = ":bl_Amyloid:pacc_time",
            tab_name = "tmp_male",
            folder = "PACC Models",
            mod_type = "PACC_GeneAb")

tmp_male$Subset <- "Males"

PACCTable_GeneSexAb_wDose$Subset <- "All"

PACCTable_GeneSexAb_bySex <- rbind(PACCTable_GeneSexAb_wDose, tmp_fem, tmp_male)
```

```{r}
# PACCTable_GeneSexAb$new_fdr_p <- p.adjust(PACCTable_GeneSexAb$p, method = "fdr")
# 
# PACCTable_GeneSexAb$fdr_p <- ifelse(is.na(PACCTable_GeneSexAb$fdr_p), PACCTable_GeneSexAb$new_fdr_p, PACCTable_GeneSexAb$fdr_p)
# PACCTable_GeneSexAb <- select(PACCTable_GeneSexAb, -c(new_fdr_p))
# 
# PACCTable_GeneSexAb_sig <- filter(PACCTable_GeneSexAb, fdr_p < 0.05)
```

```{r}
PACCTable_GeneSexAb_bySex <- PACCTable_GeneSexAb_bySex %>%
  left_join(gene_names, by = "gene") %>%
  mutate(Sig = case_when(
    p < 0.05 ~ T,
    TRUE ~ F
  ))
```

## Females {.tabset}

### Summary Table

```{r}
datatable(subset(PACCTable_GeneSexAb_bySex, Subset == "Females"),
          filter = "top",
          rownames = FALSE,
          extensions = c('Buttons'),
          options = list(dom = 'Btip',
                         buttons = c('copy', 'csv'),
                         pageLength = 15,
                         columnDefs = list(list(className = 'dt-center', targets = "_all")))) %>%
  formatRound(columns = c("beta","se", "t", "p", "fdr_p"), digits = 4) %>%
  formatStyle("fdr_p", color = styleInterval(0.05, c("red", "black")))
```

### Plotted Output

::: panel-tabset
#### ENSG00000205356

```{r}
rna$Cohort <- as.factor(rna$Cohort)
rna$SEX <- as.factor(rna$SEX)
rna$QSVERSION <- as.factor(rna$QSVERSION)


```

```{r, fig.width = 12}
fit <- lme(PACC ~ ENSG00000205356*bl_Amyloid*pacc_time + bl_pacc_age*pacc_time + EDCCNTU*pacc_time + Cohort*pacc_time + QSVERSION + Cumulative_Dose_Scaled, data = subset(rna, SEX == "Female"), random =~ pacc_time | BID, na.action = "na.omit", control = list(opt = 'optim', msMaxIter = 1000, msMaxEval = 1000))


plot_model(fit, type = "pred", terms = c("pacc_time", "ENSG00000205356", "bl_Amyloid")) +
  theme(strip.background = element_rect(fill = "white")) +
  labs(
    x = "Time (Years)",
    y = "Longitudinal PACC",
    title = bquote("Longitudinal PACC by TECPR1 and Amyloid Status in Females"),
    color = "ENSG00000205356\n(TECPR1)"
  )
```

#### ENSG00000162543

```{r, fig.width = 12}
fit <- lme(PACC ~ ENSG00000162543*bl_Amyloid*pacc_time + bl_pacc_age*pacc_time + EDCCNTU*pacc_time + Cohort*pacc_time + QSVERSION + Cumulative_Dose_Scaled, data = subset(rna, SEX == "Female"), random =~ pacc_time | BID, na.action = "na.omit", control = list(opt = 'optim', msMaxIter = 1000, msMaxEval = 1000))


plot_model(fit, type = "pred", terms = c("pacc_time", "ENSG00000162543", "bl_Amyloid")) +
  theme(strip.background = element_rect(fill = "white")) +
  labs(
    x = "Time (Years)",
    y = "Longitudinal PACC",
    title = bquote("Longitudinal PACC by UBXN10 and Amyloid Status in Females"),
    color = "ENSG00000162543\n(UBXN10)"
  )
```
:::

## Males {.tabset}

### Summary Table

```{r}
datatable(subset(PACCTable_GeneSexAb_bySex, Subset == "Males"),
          filter = "top",
          rownames = FALSE,
          extensions = c('Buttons'),
          options = list(dom = 'Btip',
                         buttons = c('copy', 'csv'),
                         pageLength = 15,
                         columnDefs = list(list(className = 'dt-center', targets = "_all")))) %>%
  formatRound(columns = c("beta","se", "t", "p", "fdr_p"), digits = 4) %>%
  formatStyle("fdr_p", color = styleInterval(0.05, c("red", "black")))
```

### Plotted Output

::: panel-tabset
#### ENSG00000279245

```{r, fig.width = 12}
fit <- lme(PACC ~ ENSG00000279245*bl_Amyloid*pacc_time + bl_pacc_age*pacc_time + EDCCNTU*pacc_time + Cohort*pacc_time + QSVERSION + Cumulative_Dose_Scaled, data = subset(rna, SEX == "Male"), random =~ pacc_time | BID, na.action = "na.omit", control = list(opt = 'optim', msMaxIter = 1000, msMaxEval = 1000))


plot_model(fit, type = "pred", terms = c("pacc_time", "ENSG00000279245", "bl_Amyloid")) +
  theme(strip.background = element_rect(fill = "white")) +
  labs(
    x = "Time (Years)",
    y = "Longitudinal PACC",
    title = bquote("Longitudinal PACC by FAM223A and Amyloid Status in Males"),
    color = "ENSG00000279245\n(FAM223A)"
  )
```

#### ENSG00000164308

```{r, fig.width = 12}
fit <- lme(PACC ~ ENSG00000164308*bl_Amyloid*pacc_time + bl_pacc_age*pacc_time + EDCCNTU*pacc_time + Cohort*pacc_time + QSVERSION + Cumulative_Dose_Scaled, data = subset(rna, SEX == "Male"), random =~ pacc_time | BID, na.action = "na.omit", control = list(opt = 'optim', msMaxIter = 1000, msMaxEval = 1000))


plot_model(fit, type = "pred", terms = c("pacc_time", "ENSG00000164308", "bl_Amyloid")) +
  theme(strip.background = element_rect(fill = "white")) +
  labs(
    x = "Time (Years)",
    y = "Longitudinal PACC",
    title = bquote("Longitudinal PACC by ERAP2 and Amyloid Status in Males"),
    color = "ENSG00000164308\n(ERAP2)"
  )
```
:::
